internal/scm/azuredevops/azuredevops.go

```diff
@@ -0,0 +1,549 @@
+package azuredevops
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/lindell/multi-gitter/internal/scm"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/core"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/git"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/identity"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v7/location"
+	"github.com/pkg/errors"
+)
+
+type AzureDevOps struct {
+	RepositoryListing
+	gitClient       git.Client
+	identityClient  identity.Client
+	locationClient  location.Client
+	coreClient      core.Client
+	SSHAuth         bool
+	pat             string
+	projectNameToID map[string]uuid.UUID
+}
+
+// RepositoryListing contains information about which repositories that should be fetched
+type RepositoryListing struct {
+	Projects     []string
+	Repositories []RepositoryReference
+}
+
+// RepositoryReference contains information to be able to reference a repository
+type RepositoryReference struct {
+	ProjectName string
+	Name        string
+}
+
+// New creates a new AzureDevOps client
+func New(pat, baseURL string, sshAuth bool, fork bool, repoListing RepositoryListing) (*AzureDevOps, error) {
+	if strings.TrimSpace(pat) == "" {
+		return nil, errors.New("Personal Access Token is empty")
+	}
+
+	if strings.TrimSpace(baseURL) == "" {
+		return nil, errors.New("base url is empty")
+	}
```

Comment:
Please do parsing of user input in `cmd/`

Reply:
It looks like the validation I was doing here is already handled in cmd/ so I'm just removing this code